### DISCUSSION TITLE
WordPress 5.8

### DISCUSSION THREAD

**Discussion by danemorgan** [holger1411/understrap#1285] [Roadmap]
This discussion is to collect anything we should be looking out for while preparing for WordPress 5.8.

To kick it off, there are new [Gutenberg Template Editor](https://make.wordpress.org/core/2021/06/16/introducing-the-template-editor-in-wordpress-5-8/) functions coming that are opt-out.

I think we should opt-out of these ahead of the 5.8 release until we are ready to support them. I don't think it's likely we will be ready to do that when 5.8 releases.

I also think we should work toward supporting them as soon as we can.

**Comment by LukaszJaro** [comment]
- Watch out for the new blocks in widget areas, I had experiences where my widgets would be deleted after I registered new widget area, so some testing around that

- Performance enhancement: Add block-styles loading enhancements for classic themes? https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/

This is how I tried the implementation on another theme but as article mentioned you need to watch out for style priorities:
```
//loads separate stylesheets for each block on page
add_filter( 'load_separate_block_assets', '__return_true' );
//inlines the CSS when stylesheets are of smaller nature
add_action( 'wp_footer', 'gutenberg_maybe_inline_styles', 1 );
```

> ...style priorities. Opting-in to separate styles loading in a classic theme means that the loading order of styles changes. Block styles that used to be in the head will move to the footer, so you will need to check your theme’s styles and make sure any opinionated styles you add to blocks have a higher priority than core styles.

- WordPress is dropping IE11, so I think understrap should remove any special ie11 support(css, scripts, browserslistrc, etc). Bootstrap 5 also drops ie11 support.

**Reply by bacoords** [reply]
Thanks for these notes! Here's some thoughts to continue the discussion.

> Watch out for the new blocks in widget areas, I had experiences where my widgets would be deleted after I registered new widget area, so some testing around that

We'll keep an eye out. We're adding more Gutenberg support in the next update.

> Performance enhancement: Add block-styles loading enhancements for classic themes? https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/

We've been working on adding better block styling overall, so you'll see that in the new update coming soon (or on the `develop` branch now. 

> WordPress is dropping IE11, so I think understrap should remove any special ie11 support(css, scripts, browserslistrc, etc). Bootstrap 5 also drops ie11 support.

I believe the plan is to keep IE11 support as long as we're on BS4, which includes it and is part of it. Plus we'd have to change our minimum WP version to 5.8, which might feel a little too soon now. When we migrate to BS5, it will probably make sense to say we've dropped support for IE11.

**Reply by LukaszJaro** [reply]
Here's more solutions for the block loading enhancements:

If you have an existing theme and you want to opt-in to this improvement, you will need to test your theme for style priorities. Opting-in to separate styles loading in a classic theme means that the loading order of styles changes. Block styles that used to be in the head will move to the footer, so you will need to check your theme’s styles and make sure any opinionated styles you add to blocks have a higher priority than core styles.

There are many ways you can address this on your theme, it all comes down to what you want to do.

- **Solution 1**: Increasing specificity for block CSS in the theme. You can change the specificity of theme theme’s CSS rules so they override the default block styles rules. If for example the theme was previously using .wp-block-button as a selector in its CSS, it can now change to :root .wp-block-button, body .wp-block-button, .wp-block-button.wp-block-button or anything else that has a higher specificity than the .wp-block-button selector.

- **Solution 2**: Making the theme’s blocks stylesheet load after core styles: You can change the way the theme’s styles get loaded. Instead of hooking in wp_enqueue_scripts you can hook in wp_footer. This will make the theme’s stylesheet load after the block styles have been added, and therefore override them. Note: This should be avoided if you don’t have separate stylesheets for block-styles and global styles.

- **Solution 3**: Don’t use separate styles loading. Use this solution if your theme is not ready for these improvements, or if you don’t want to deal with the complexities of the above 2 solutions. If you have added add_filter( 'should_load_separate_core_block_assets', '__return_true' ); in your theme remove it. If you haven’t added that line, the add add_filter( 'should_load_separate_core_block_assets', '__return_false' ); and that should disable the feature.

The above only makes sense if you are not working on a block-based theme. If you are on a block-based theme, then something else must be wrong and should be further investiated.

https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/
